launchcontainer/launchcontainer.py

```diff
@@ -143,82 +171,36 @@ class LaunchContainerXBlock(XBlock, xblocksettings.XBlockWithSettingsMixin):
-        except (Site.DoesNotExist, AttributeError):  # Probably on the lms side.
-            if get_current_site:
-                site = get_current_site()  # From the request.
-            else:
-                site = Site.objects.all().order_by('domain').first()
-
-        url = cache.get(make_cache_key(site.domain))
-        if url:
-            return url
-
-        # Nothing in the cache. Go find the URL.
-        site_wharf_url = None
-        if hasattr(site, 'configuration'):
-            site_wharf_url = site.configuration.get_value(WHARF_URL_KEY)
-        elif siteconfig_helpers:
-            # Rely on edX's helper, which will fall back to the microsites app.
-            site_wharf_url = siteconfig_helpers.get_value(WHARF_URL_KEY)
-
-        urls = (
-            # A SiteConfig object: this is the preferred implementation.
-            (
-                'SiteConfiguration',
-                site_wharf_url
-            ),
-            # A string: the currently supported implementation.
-            (
-                "ENV_TOKENS[{}]".format(WHARF_URL_KEY),
-                settings.ENV_TOKENS.get(WHARF_URL_KEY)
-            ),
-            # A dict: the deprecated version.
-            (
-                "ENV_TOKENS['LAUNCHCONTAINER_API_CONF']",
-                settings.ENV_TOKENS.get('LAUNCHCONTAINER_API_CONF', {}).get('default')
-            ),
-        )
+    def get_wharf_url(self, required=True):
+        """
+        Get WHARF URL.
+        """
+        course_id = getattr(self, "course_id", None)
 
-        try:
-            url = next((x[1] for x in urls if is_valid(x[1])))
-        except StopIteration:
-            raise ImproperlyConfigured("No Virtual Labs URL was found, "
-                                       "please contact your site administrator.")
+        # System-wide fallback settings
+        system_default_wharf_url = settings.ENV_TOKENS.get(WHARF_URL_KEY)
+
+        # Attempt getting Tahoe site-specific WHARF URL
+        wharf_url = get_tahoe_wharf_url_for_course(
+            course_id, default=system_default_wharf_url
+        )
 
-        if not url:
-            raise AssertionError("You must set a valid url for the launchcontainer XBlock. "
-                                 "URLs attempted: {}".format(urls)
-                                 )
+        message = "A valid url for the LaunchContainer XBlock is needed: {}".format(wharf_url)
 
-        cache.set(make_cache_key(site), url, CACHE_KEY_TIMEOUT)
+        if wharf_url:
+            if not is_valid(wharf_url):
+                raise ImproperlyConfigured(message)  # Invalid URL
+        elif required:
+            raise ImproperlyConfigured(message)  # Missing required URL
 
-        logger.debug("XBlock-launchcontainer urls attempted: {}".format(urls))
+        return wharf_url
 
-        return url
+    def get_wharf_delete_url(self, required=True):
+        wharf_url = self.get_wharf_url(required=required)
```

Comment:
I'm not sure I totally understand why we check for a wharf url and then return None.  I *think* it's because we want this method (`get_wharf_delete_url`) to also make it clear that no URL is available.  I'm also not sure why we'd return a delete URL if `required=False` but there is no `wharf_url`.  I just need a docstring or comment here, I think.